### ISSUE TITLE
Changing data on an existing chart without calling dispose() breaks responsivity

### ISSUE THREAD

**Issue by Muntaner** [amcharts/amcharts4#2184]
I didn't test if this issue is occurring on all chart types, but I could reproduce [a minimal CodePen example with the issue using a PieChart](https://codepen.io/Muntaner/pen/jOPmbgY). Still unclear if this is a problem related to _createFromConfig_ or it's always happening.

In short:
- create a chart instance with some data and basic series/configuration, using _am4core.createFromConfig_. The charts works flawlessly and it's 100% responsive
- after sometime (I used a setTimeout in the CodePen) override the data in the configuration, by setting it on the variable _config.data = [...]_ . Now, instance again the chart by calling _am4core.createFromConfig_. Data is correctly updated and user interaction still works, but **the chart and the legend are no longer responsive**.
- this can be fixed by executing chart.dispose() before overriding the data (commented in the attached CodePen)

Tested both on Chrome and Firefox.

**Comment by martynasma** [comment]
Why would you recreate the whole chart if you just updating data? It's too much of wasted overhead.

Just update chart's data:

```
chart.data = [
    {
      country: "Country 4",
      value: 500
    },
    {
      country: "Country 5",
      value: 300
    }
  ];
```

The chart will automatically update itself for the new data.

And it will be much faster, and won't break anything :)

**Comment by Muntaner** [comment]
Yes, I 100% agree that the example I provided may look like a useless overkill and isn't the best way to update an existing chart.

Btw, I discovered the issue in a more complex example where I was recalculating the data, the series and other parameters, and re-creating the chart from the scratch looked like a good idea. What you see in the example is just a stripped-down-to-the-minimum scenario reproducing the problem. ;)
